go/tasks/plugins/array/k8s/transformer.go

```diff
@@ -94,6 +94,40 @@ func buildPodMapTask(task *idlCore.TaskTemplate, metadata core.TaskExecutionMeta
 	return pod, nil
 }
 
+// Here we customize the k8sPod primary container by templatizing args.
+// The call to ToK8sPodSpec for the task container target
+// case already handles this but we must explicitly do so for K8sPod task targets.
+func modifyMapPodTaskPrimaryContainer(ctx context.Context, tCtx core.TaskExecutionContext, arrTCtx *arrayTaskContext, container *v1.Container) error {
+	var err error
```

Comment:
isnt this container modification a simple shared library (flytek8s)

Reply:
it is, but for flyte container definitions, not k8s ones. ideally all this duplicate code[ goes away](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/plugins/array/k8s/transformer.go#L56) after we introduce array nodes

Reply:
Wild guess: are you, @kumare3, asking about `consistency` as in `concurrent programming consistency` ? it took me a sec to realize how off base I was :-D

Reply:
what?